### PR TITLE
use calico's python-etcd fork, may fix #1092 for 1.4 branch.

### LINKED ISSUE #1092: Fix incompatibility with etcd3 server v2 api
Calico uses a Python etcd client.  It is currently incompatible with the etcd3 server.  The Etcd3 server correctly enforces best security practices and refuses connections from TLSv1. 

Root cause issue is documented here:
https://github.com/jplana/python-etcd/pull/196

and
https://github.com/projectcalico/calico-containers/issues/1107

### PR DESCRIPTION
**Pull request by vixns**
Allow felix 1.4.x to connect to etcd+tls, not fully tested yet.

### CODE REVIEW

**Comment by marvin-tigera** [comment]
Can one of the admins verify this patch?

File: felix_requirements.txt [hunk 1]
```diff
@@ -1,7 +1,6 @@
 gevent
 greenlet
 netaddr
-python-etcd>=0.4.1
```
Comment (fasaxc) [inline felix_requirements.txt]: I think this requirement should still be here.

File: pyi/Dockerfile [hunk 2]
```diff
@@ -16,7 +16,7 @@ RUN ./build-prereqs.sh
 ENV LD_LIBRARY_PATH=/usr/local/lib
 RUN yum install -y libffi-devel libffi yajl
 ADD felix_requirements.txt felix_requirements.txt
-RUN pip install -U -r felix_requirements.txt
```
Comment (fasaxc) [inline pyi/Dockerfile]: If you split this into two `pip` commands and run the `python-etcd` one first then you should get the github version of python-etcd.

Should we pin this to a particular commit for repeatability?

Why do you add -e here?  That installs python-etcd as a development egg, which shouldn't be needed.

**Comment by marvin-tigera** [comment]
Can one of the admins verify this patch?

**Comment by vixns** [comment]
I've tried two pip commands as suggested, but still some warnings.
Anyway, I've upgraded to 2.0, TLS support just works, thanks for the advice.

I no longer use the 1.4 branch, I'm closing this PR.

**Pull request closed without merging**